### PR TITLE
Update to embedded-hal 1.0.

### PR DESCRIPTION
**Pull request by qwandor**
The biggest change here is that the new SpiDevice trait explicitly manages the CS pin, so we never need to manage it manually.

### CODE REVIEW

File: examples/low_power_lab_receive.rs [hunk 1]
```diff
@@ -12,20 +14,19 @@ fn main() -> Result<()> {
...
     let options = SpidevOptions::new()
         .bits_per_word(8)
         .max_speed_hz(1_000_000)
         .mode(SpiModeFlags::SPI_MODE_0)
         .build();
-    spi.configure(&options)?;
+    spi_bus.configure(&options)?;
+    let spi_bus = Mutex::new(spi_bus);
+
+    let spi = MutexDevice::new(&spi_bus, cs, Delay);
```
Comment (almusil) [inline examples/low_power_lab_receive.rs:26]: I don't think there is a need to complicate things with `Mutex`, it should be enough to use `ExclusiveDevice`.

File: examples/low_power_lab_send.rs [hunk 2]
```diff
@@ -12,20 +14,19 @@ fn main() -> Result<()> {
...
     let options = SpidevOptions::new()
         .bits_per_word(8)
         .max_speed_hz(1_000_000)
         .mode(SpiModeFlags::SPI_MODE_0)
         .build();
-    spi.configure(&options)?;
+    spi_bus.configure(&options)?;
+    let spi_bus = Mutex::new(spi_bus);
+
+    let spi = MutexDevice::new(&spi_bus, cs, Delay);
```
Comment (almusil) [inline examples/low_power_lab_send.rs:26]: Same as above.

File: examples/receive.rs [hunk 3]
```diff
@@ -1,18 +1,12 @@
...
 use linux_embedded_hal::spidev::{SpiModeFlags, SpidevOptions};
-use linux_embedded_hal::sysfs_gpio::Direction;
-use linux_embedded_hal::{Spidev, SysfsPin};
+use linux_embedded_hal::{SpidevDevice, SysfsPin};
 use rfm69::Rfm69;
 use utilities::rfm_error;
 
 fn main() -> Result<()> {
-    // Configure CS pin
-    let cs = SysfsPin::new(25);
```
Comment (almusil) [inline examples/receive.rs:10]: There is missed `SysfsPin` usage at line 40 (34 after change).

File: examples/send.rs [hunk 4]
```diff
@@ -1,18 +1,12 @@
...
 use linux_embedded_hal::spidev::{SpiModeFlags, SpidevOptions};
-use linux_embedded_hal::sysfs_gpio::Direction;
-use linux_embedded_hal::{Spidev, SysfsPin};
+use linux_embedded_hal::{SpidevDevice, SysfsPin};
 use rfm69::Rfm69;
 use utilities::rfm_error;
 
 fn main() -> Result<()> {
-    // Configure CS pin
-    let cs = SysfsPin::new(25);
```
Comment (almusil) [inline examples/send.rs:10]: Same as `receive.rs`.

**Review by almusil — CHANGES_REQUESTED** [review]
Hi,

thank you for the PR. This is definitely an improvement. The change seems to be fine overall I have some minor comments inline.

File: examples/low_power_lab_receive.rs (hunk 1 above)
Comment (qwandor) [reply examples/low_power_lab_receive.rs]: Done.

File: examples/low_power_lab_send.rs (hunk 2 above)
Comment (qwandor) [reply examples/low_power_lab_send.rs]: Done.

File: examples/receive.rs (hunk 3 above)
Comment (qwandor) [reply examples/receive.rs:10]: The idea was to demonstrate how to use `SpidevDevice` in this example, where the device driver takes care of managing the associated CS pin automatically.

Comment (almusil) [reply examples/receive.rs:10]: Right that's fine, what I meant is that this bit is not needed then `SysfsPin::new(25).unexport()?;`

Comment (qwandor) [reply examples/receive.rs:10]: Ah right sorry, I misread your comment. Fixed.

File: examples/send.rs (hunk 4 above)
Comment (qwandor) [reply examples/send.rs:10]: Done.

**Review by almusil — APPROVED** [review]
Thanks!

**Pull request merged**